### PR TITLE
[TT-5577] strip listen path by default on OAS import

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description

strip listen path while importing OAS doc.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

**Review by urbanishimwe — APPROVED** [review]

**Review by titpetric — APPROVED** [review]

File: gateway/api.go [hunk 1]
```diff
@@ -2835,6 +2835,8 @@ func (gw *Gateway) makeImportedOASTykAPI(next http.HandlerFunc) http.HandlerFunc
 			return
 		}
 
+		oasObj.GetTykExtension().Server.ListenPath.Strip = true
```
Comment (furkansenharputlu) [inline gateway/api.go:2838]: I think this should be inside BuildDefaultTykExtension.

Comment (jeffy-mathew) [reply gateway/api.go:2838]: We use buildDefaultTykExtension is also used in patch endpoint

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4112/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2447500435)

File: gateway/api.go (hunk 1 above)
Comment (furkansenharputlu) [reply gateway/api.go:2838]: We can pass a param. Building the extension is that function’s job.

Comment (furkansenharputlu) [reply gateway/api.go:2838]: “If post” check should work

Comment (jeffy-mathew) [reply gateway/api.go:2838]: Actually, it'd make sense pass a param to handle some other flags as well.
Internal/external, active etc.

Will make those changes as well

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4112/merge
Commit: 385bc5e3301cc66c0882da1e88b54bf739dd3057 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2447717259)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4112/merge
Commit: aff825ae807d8d7418ae34c8de85435a3533f91a 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2452769796)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4112/merge
Commit: ae46eed65bb1f369c8dfbff3c247186c1f27e32b 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2453978870)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 47999b7dd15458a9a37a535b786b6c157bea1a3b [TT-5577] strip listen path by default on OAS import (#4112)

* strip listen path by default on OAS import
* add flag isImport on BuildDefaultTykExtension

[changelog]
added: strip listen path while importing OAS doc.
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2454117938)

**Pull request merged**